.builder/actions/mock_server_setup.py

```diff
@@ -0,0 +1,40 @@
+"""
+Setup local mock server for tests
+"""
+
+import Builder
+
+import os
+import subprocess
+
+
+class MockServerSetup(Builder.Action):
+    """
+    Set up this machine for running the mock server test
+
+    This action should be run in the 'pre_build_steps' or 'build_steps' stage.
+    """
+
+    def run(self, env):
+        self.env = env
+        if os.name == 'nt':
+            python_execute = "python"
+        else:
+            python_execute = "python3"
+
+        # install dependency for mock server
+        result = self.env.shell.exec(python_execute,
+                                     '-m', 'install', 'h11', 'trio')
+        if result.returncode != 0:
+            print("Mock server failed to setup, skip the mock server tests.")
+            return
+
+        # set cmake flag so mock server tests are enabled
+        env.project.config['cmake_args'].append(
+            '-DENABLE_MOCK_SERVER_TESTS=ON')
+
+        base_dir = os.path.dirname(os.path.realpath(__file__))
+        dir = os.path.join(base_dir, "..", "..", "tests", "mock_s3_server")
+        os.chdir(dir)
+
+        subprocess.Popen([python_execute, "mock_s3_server.py"])
```

Comment:
add a comment that this is launching a background process that runs forever

Reply:
maybe we could do something to try and clean this up afterwards?

maybe register an [atexit](https://docs.python.org/3/library/atexit.html) handler that tries to kill the background process?